node/src/daemon/setup_reporter.rs

```diff
@@ -119,7 +119,7 @@ impl SetupReporter for SetupReporterReal {
                 .populate(self.dirs_wrapper.as_ref())
         });
         let data_directory = match all_but_configured.get("data-directory") {
-            Some(uisrv) if uisrv.status == Set => PathBuf::from(&uisrv.value),
+            Some(uisrv) if uisrv.status == Set => PathBuf::from(&uisrv.value), // TODO drive in the specific chain dir here
```

Comment:
First run all tests together to see them all are passing. Use todo!() after the arrow (instead of PathBuf::from(&uisrv.value)). If you run all tests again and no test is failing that means this arm is not properly tested. Then you would have to write a test. 

If it is tested just remove the TODO comment.